apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/components/AssessmentQuestionTable.tsx

```diff
@@ -0,0 +1,798 @@
+    'status',
+    parseAsArrayOf(parseAsStringLiteral(GRADING_STATUS_VALUES)).withDefault(
+      DEFAULT_GRADING_STATUS_FILTER,
+    ),
+  );
+  const [assignedGraderFilter, setAssignedGraderFilter] = useQueryState(
+    'assigned_grader',
+    parseAsArrayOf(parseAsString).withDefault(DEFAULT_ASSIGNED_GRADER_FILTER),
+  );
+  const [gradedByFilter, setGradedByFilter] = useQueryState(
+    'graded_by',
+    parseAsArrayOf(parseAsString).withDefault(DEFAULT_GRADED_BY_FILTER),
+  );
+  const [submissionGroupFilter, setSubmissionGroupFilter] = useQueryState(
+    'submission_group',
+    parseAsArrayOf(parseAsString).withDefault(DEFAULT_SUBMISSION_GROUP_FILTER),
+  );
+  const [aiAgreementFilter, setAiAgreementFilter] = useQueryState(
+    'ai_agreement',
+    parseAsArrayOf(parseAsString).withDefault(DEFAULT_AI_AGREEMENT_FILTER),
+  );
+  const [rubricItemsFilter, setRubricItemsFilter] = useQueryState(
+    'rubric_items',
+    parseAsArrayOf(parseAsString).withDefault([]),
+  );
+
+  const [manualPointsFilter, setManualPointsFilter] = useQueryState(
+    'manual_points',
+    parseAsNumericFilter.withDefault(''),
+  );
+  const [autoPointsFilter, setAutoPointsFilter] = useQueryState(
+    'auto_points',
+    parseAsNumericFilter.withDefault(''),
+  );
+  const [totalPointsFilter, setTotalPointsFilter] = useQueryState(
+    'total_points',
+    parseAsNumericFilter.withDefault(''),
+  );
+  const [scoreFilter, setScoreFilter] = useQueryState(
+    'score',
+    parseAsNumericFilter.withDefault(''),
+  );
+
+  const { createCheckboxProps } = useShiftClickCheckbox<InstanceQuestionRow>();
+
+  const [columnSizing, setColumnSizing] = useState<ColumnSizingState>({});
+  const [rowSelection, setRowSelection] = useState({});
+  const [showDeleteAiGradingModal, setShowDeleteAiGradingModal] = useState(false);
+  const [showDeleteAiGroupingsModal, setShowDeleteAiGroupingsModal] = useState(false);
+  const scrollRef = useRef<HTMLDivElement | null>(null);
+
+  const queryClientInstance = useQueryClient();
+
+  // Fetch instance questions data
+  const {
+    data: instanceQuestionsInfo = initialInstanceQuestionsInfo,
+    error: instanceQuestionsError,
+    isError: isInstanceQuestionsError,
+  } = useQuery<InstanceQuestionRow[]>({
+    queryKey: ['instance-questions'],
+    queryFn: async () => {
+      const res = await fetch(window.location.pathname + '/instances.json', {
+        headers: {
+          Accept: 'application/json',
+        },
+      });
+      const data = await res.json();
+      if (!res.ok) {
+        throw new Error(data.error);
+      }
+      if (!data.instance_questions) throw new Error('Invalid response format');
+      const parsedData = z.array(InstanceQuestionRowSchema).safeParse(data.instance_questions);
+      if (!parsedData.success) throw new Error('Failed to parse instance questions');
+      return parsedData.data;
+    },
+    refetchInterval: false,
+    initialData: initialInstanceQuestionsInfo,
```

Comment:
If I'm reading the docs correctly, our lack of other complementary config means that we'll immediately refetch all this data on page load, which seems wasteful: https://tanstack.com/query/latest/docs/framework/react/guides/initial-query-data#staletime-and-initialdataupdatedat